asr1k_neutron_l3/models/neutron/l3/access_list.py

```diff
@@ -30,9 +33,43 @@ def _rest_definition(self):
         acl = access_list.AccessList(name=self.id, drop_on_17_3=self._drop_on_17_3)
         for i, rule in enumerate(self.rules):
             sequence = (i + 1) * 10
-            ace_rule = access_list.ACERule(access_list=self.id, acl_rule=sequence, action=rule.action,
-                                           protocol=rule.protocol, ipv4_address=rule.source, mask=rule.source_mask,
-                                           dest_ipv4_address=rule.destination, dest_mask=rule.destination_mask)
+
+            ip_args = dict()
+            if rule.source_mask:
+                ip_args['ipv4_address'] = rule.source
+                ip_args['mask'] = rule.source_mask
+            else:
+                ip_args['host'] = rule.source
+
+            if rule.destination_mask:
+                ip_args['dest_ipv4_address'] = rule.destination
+                ip_args['dest_mask'] = rule.destination_mask
+            else:
+                ip_args['dst_host'] = rule.destination
```

Comment:
I have know hidden the `port_args` and `ip_args` creation in the `Rule` object. This could all be a lot simpler if `netconf_yang.access_list.ACERule` and `ACLRule` would not need the `id` aka the ACL name. I believe they also do not need it, but I see that more to be the task of a follow up.